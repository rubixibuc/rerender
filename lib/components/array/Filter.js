import React from "react";
import PropTypes from "prop-types";

const render = ({ children, filter, items }) =>
  (items || null) && children(items.filter(filter));

render.propTypes = {
  children: PropTypes.func.isRequired,
  filter: PropTypes.func.isRequired,
  items: PropTypes.array
};

export default render;
