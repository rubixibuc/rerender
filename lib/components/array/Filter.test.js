import React from "react";
import Filter from "./Filter";
import renderer from "react-test-renderer";

test("should render nothing when items is undefined", () => {
  const component = renderer.create(
    <Filter filter={i => i} items={undefined}>
      {items => <div>{items}</div>}
    </Filter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render nothing when items is null", () => {
  const component = renderer.create(
    <Filter filter={i => i} items={null}>
      {items => <div>{items}</div>}
    </Filter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render nothing when items is empty", () => {
  const component = renderer.create(
    <Filter filter={i => i} items={[]}>
      {items => <div>{items}</div>}
    </Filter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("should render filtered items is when items is not empty", () => {
  const component = renderer.create(
    <Filter filter={i => i > 1} items={[1, 2]}>
      {items => <div>{items}</div>}
    </Filter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
