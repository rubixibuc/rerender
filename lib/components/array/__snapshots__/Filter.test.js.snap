// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render filtered items is when items is not empty 1`] = `
<div>
  2
</div>
`;

exports[`should render nothing when items is empty 1`] = `<div />`;

exports[`should render nothing when items is null 1`] = `null`;

exports[`should render nothing when items is undefined 1`] = `null`;
